common/transmitmedia.cpp

```diff
@@ -714,15 +717,16 @@ bytevector SrtSource::Read(size_t chunk)
         data.resize(chunk);
 
     CBytePerfMon perf;
-    srt_bstats(m_sock, &perf, true);
-    if ( transmit_bw_report && int(counter % transmit_bw_report) == transmit_bw_report - 1 )
+    srt_bstats(m_sock, &perf, clear_stats);
+    clear_stats = false;
+    if ( transmit_bw_report && (counter % transmit_bw_report) == transmit_bw_report - 1 )
     {
         cout << "+++/+++SRT BANDWIDTH: " << perf.mbpsBandwidth << endl;
     }
-
-    if ( transmit_stats_report && counter % transmit_stats_report == transmit_stats_report - 1)
+    if ( transmit_stats_report && (counter % transmit_stats_report) == transmit_stats_report - 1)
     {
         PrintSrtStats(m_sock, perf);
+        clear_stats = !total_stats;
```

Comment:
Shouldn't you simply pass `!total_stats` instead of `true` in the call of `srt_bstats`? The stats are being cleared **after extraction**, so you shouldn't need any proxy.

Reply:
Hi @ethouris ,

> Shouldn't you simply pass !total_stats instead of true in the call of srt_bstats? The stats are being cleared after extraction, so you shouldn't need any proxy.

You're right! I do in this way because my first attempt has two setters for clear_stats. With the last implementation (the commited), this isn't true.

I'll prepare a refactoring! Thank you!

Reply:
Hi @ethouris ,

Sorry! No, no, no!
The call to the `srt_bstats()` clearing the data needs to be done **ONLY** when the Statistics are printed. So, you can call each time with true or false. You need to call **EVERY** time with `false` and only one time with `true`. And, as you can clear **after extraction**, you need to save the state and clear for the **next** call.

So, this doesn't need to be fixed. It's correct!